osmo_jupyter/dataset.py

```diff
@@ -0,0 +1,267 @@
+    """
+        Parse a list of timestamped equilibration statuses into a DataFrame of start
+        and end times of equilibrated states. Assumes all time between unequilibrated
+        and equilibrated input points are unequilibrated.
+
+        Args:
+            equilibration_status: A Series of equilibration status strings.
+        Returns:
+            DataFrame of start and end times of equilibrated states.
+    """
+    # Get a boolean series with timestamped index of equilbrated data points
+    equilibrated_mask = equilibration_status == "equilibrated"
+    leading_edges = equilibrated_mask[equilibrated_mask.astype(int).diff() == 1].index
+    trailing_edges = equilibrated_mask[
+        equilibrated_mask.astype(int).diff(periods=-1) == 1
+    ].index
+
+    # Trim edges to align
+    if len(leading_edges) > len(trailing_edges):
+        leading_edges = leading_edges[:-1]
+    if len(trailing_edges) > len(leading_edges):
+        trailing_edges = trailing_edges[1:]
+
+    return pd.DataFrame({"start_time": leading_edges, "end_time": trailing_edges})
+
+
+def pivot_process_experiment_results_on_ROI(
+    experiment_df: pd.DataFrame,
+    ROI_names: List[str] = None,
+    msorm_types: List[str] = ["r_msorm", "g_msorm", "b_msorm"],
+) -> pd.DataFrame:
+    """
+        Flatten a DataFrame of process experiment results down to one row per image.
+
+        Args:
+            experiment_df: A DataFrame of process experiment summary statistics.
+            ROI_names: Optional. A list of ROI names to select. Defaults to all ROIs present.
+            msorm_types: Optional. A list of MSORM column names to select. Defaults to all RGB channel MSORMs.
+        Returns:
+            DataFrame with one row per image, and msorm values for a subset of ROIs.
+    """
+    if ROI_names:
+        experiment_df = experiment_df[experiment_df["ROI"].isin(ROI_names)]
+
+    # Pivot creates a heirachical multiindex data frame with an index for each 'values' column
+    pivot = experiment_df.pivot(
+        index="timestamp", columns="ROI", values=msorm_types + ["image"]
+    )
+
+    # There's one copy of the image name for each ROI due to the pivot, so pull out the top level index
+    images = pivot["image"][experiment_df["ROI"].values[0]]
+    pivot.drop("image", axis=1, inplace=True)
+    # Flatten the pivot table index, and add images back in
+    pivot.columns = [" ".join(col[::-1]).strip() for col in pivot.columns.values]
+    pivot["image"] = images
+
+    return pivot
+
+
+def open_and_combine_process_experiment_results(
+    process_experiment_result_filepaths: List[str],
+    ROI_names: List[str] = None,
+    msorm_types: List[str] = ["r_msorm", "g_msorm", "b_msorm"],
+) -> pd.DataFrame:
+    """
+        Open multiple process experiment result files and combine into a single DataFrame with one row per image.
+
+        Args:
+            process_experiment_result_filepaths: A list of filepaths to process experiment summary statistics files.
```

Comment:
Not sure what'll happen if ROIs don't match; do you know? I'm fine leaving it explicitly undefined
```suggestion
            process_experiment_result_filepaths: A list of filepaths to process experiment summary statistics files.
                These are all expected to have the same ROIs.
```

Reply:
Oh, I think it will just produce a bunch on NaN values. I can verify, but I'll add a comment about that.